helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloLogService.java

```diff
@@ -0,0 +1,104 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.testing;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static com.spotify.docker.client.DockerClient.LogsParam.follow;
+import static com.spotify.docker.client.DockerClient.LogsParam.stderr;
+import static com.spotify.docker.client.DockerClient.LogsParam.stdout;
+
+import com.spotify.docker.client.DockerClient;
+import com.spotify.docker.client.LogStream;
+import com.spotify.helios.common.descriptors.HostStatus;
+import com.spotify.helios.common.descriptors.JobId;
+import com.spotify.helios.common.descriptors.TaskStatus;
+
+import com.google.common.collect.Maps;
+import com.google.common.util.concurrent.AbstractScheduledService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+class HeliosSoloLogService extends AbstractScheduledService {
+
+  private static final Logger log = LoggerFactory.getLogger(HeliosSoloLogService.class);
+
+  private final HeliosSoloDeployment solo;
+  private final DockerClient dockerClient;
+
+  private Map<String, ScheduledFuture<?>> logFutures = Maps.newHashMap();
+
+  HeliosSoloLogService(final HeliosSoloDeployment solo, final DockerClient dockerClient) {
+    this.solo = solo;
+    this.dockerClient = dockerClient;
+  }
+
+  @Override
+  protected void runOneIteration() throws Exception {
+    try {
+      // fetch all the jobs running on the solo deployment
+      final HostStatus hostStatus = solo.client().hostStatus(solo.agentName()).get();
+      final Map<JobId, TaskStatus> statuses = hostStatus.getStatuses();
+
+      for (final TaskStatus status : statuses.values()) {
+        final JobId jobId = status.getJob().getId();
+        final String containerId = status.getContainerId();
+        if (isNullOrEmpty(containerId) || logFutures.containsKey(containerId)) {
+          continue;
+        }
+
+        // for any containers we're not already tracking, schedule attaching to their stdout/stderr
+        final ScheduledFuture<?> future = this.executor().schedule(new Runnable() {
+          @Override
+          public void run() {
+            try {
+              final LogStream logStream = dockerClient.logs(containerId, stdout(), stderr(),
+                                                            follow());
+              log.info("attaching stdout/stderr for job={}, container={}", jobId, containerId);
+              logStream.attach(System.out, System.err);
+            } catch (final Exception e) {
+              log.warn("error streaming log for job={}, container={} - {}", jobId, containerId, e);
+            }
+          }
+        }, 100, TimeUnit.MILLISECONDS);
```

Comment:
not very important, but - does the delay matter at all? Is it enough to call `executor.submit(runnable)` and just know that the task will be executed asynchronously from this Service / run loop? (i.e. whenever the executor gets to it)